<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Disease analysis | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="Chapter 4 Disease analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Disease analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Disease analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter3.html">
<link rel="next" href="chapter5.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.6/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#terminology"><i class="fa fa-check"></i><b>1.1</b> Terminology</a><ul>
<li class="chapter" data-level="1.1.1" data-path="chapter1.html"><a href="chapter1.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>1.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="1.1.2" data-path="chapter1.html"><a href="chapter1.html#gene-ontology-go"><i class="fa fa-check"></i><b>1.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="1.1.3" data-path="chapter1.html"><a href="chapter1.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>1.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="1.1.4" data-path="chapter1.html"><a href="chapter1.html#other-gene-sets"><i class="fa fa-check"></i><b>1.1.4</b> Other gene sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Functional Enrichment Analysis Methods</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#over-representation-analysis"><i class="fa fa-check"></i><b>2.1</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.3</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#input-data"><i class="fa fa-check"></i><b>3.1</b> Input data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#cell-marker"><i class="fa fa-check"></i><b>3.3</b> Cell Marker</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#msigdb-analysis"><i class="fa fa-check"></i><b>3.4</b> MSigDb analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="chapter4.html"><a href="chapter4.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Gene Ontology Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter5.html"><a href="chapter5.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="chapter5.html"><a href="chapter5.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chapter5.html"><a href="chapter5.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>5.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="chapter12.html"><a href="chapter12.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="chapter12.html"><a href="chapter12.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="chapter12.html"><a href="chapter12.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="chapter12.html"><a href="chapter12.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="chapter12.html"><a href="chapter12.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> dplyr verbs for clusterProfiler</a><ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#filter"><i class="fa fa-check"></i><b>13.1</b> filter</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#arrange"><i class="fa fa-check"></i><b>13.2</b> arrange</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#select"><i class="fa fa-check"></i><b>13.3</b> select</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#mutate"><i class="fa fa-check"></i><b>13.4</b> mutate</a></li>
<li class="chapter" data-level="13.5" data-path="chapter13.html"><a href="chapter13.html#summarise"><i class="fa fa-check"></i><b>13.5</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="chapter14.html"><a href="chapter14.html"><i class="fa fa-check"></i><b>14</b> Useful utilities</a><ul>
<li class="chapter" data-level="14.1" data-path="chapter14.html"><a href="chapter14.html#bitr"><i class="fa fa-check"></i><b>14.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="14.1.1" data-path="chapter14.html"><a href="chapter14.html#bitr_kegg"><i class="fa fa-check"></i><b>14.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="chapter14.html"><a href="chapter14.html#setReadable"><i class="fa fa-check"></i><b>14.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter4" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Disease analysis</h1>
<p><a href="https://www.bioconductor.org/packages/DOSE">DOSE</a><span class="citation">(Yu et al. <a href="#ref-yu_dose_2015">2015</a>)</span> supports Disease Ontology (DO) Semantic and Enrichment analysis. The <code>enrichDO</code> function is very useful for identifying disease association of interesting genes, and function <code>gseDO</code> function is designed for gene set enrichment analysis of <em>DO</em>.</p>
<p>In addition, <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a> also supports enrichment analysis of <a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (NCG)<span class="citation">(A. et al. <a href="#ref-omer_ncg">2016</a>)</span> and <a href="http://disgenet.org/">Disease Gene Network</a><span class="citation">(Janet et al. <a href="#ref-janet_disgenet">2015</a>)</span>, please refer to the <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a> vignettes.</p>
<div id="enrichdo-function" class="section level2">
<h2><span class="header-section-number">4.1</span> <code>enrichDO</code> function</h2>
<p>In the following example, we selected fold change above 1 as the differential genes and analyzing their disease association.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DOSE)
<span class="kw">data</span>(geneList)
gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="fl">1.5</span>]
<span class="kw">head</span>(gene)</code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">enrichDO</span>(<span class="dt">gene          =</span> gene,
              <span class="dt">ont           =</span> <span class="st">&quot;DO&quot;</span>,
              <span class="dt">pvalueCutoff  =</span> <span class="fl">0.05</span>,
              <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,
              <span class="dt">universe      =</span> <span class="kw">names</span>(geneList),
              <span class="dt">minGSSize     =</span> <span class="dv">5</span>,
              <span class="dt">maxGSSize     =</span> <span class="dv">500</span>,
              <span class="dt">qvalueCutoff  =</span> <span class="fl">0.05</span>,
              <span class="dt">readable      =</span> <span class="ot">FALSE</span>)
<span class="kw">head</span>(x)</code></pre></div>
<pre><code>##                    ID                    Description
## DOID:170     DOID:170         endocrine gland cancer
## DOID:10283 DOID:10283                prostate cancer
## DOID:3459   DOID:3459               breast carcinoma
## DOID:3856   DOID:3856 male reproductive organ cancer
## DOID:824     DOID:824                  periodontitis
## DOID:3905   DOID:3905                 lung carcinoma
##            GeneRatio  BgRatio       pvalue    p.adjust
## DOID:170      48/331 472/6268 5.662129e-06 0.004784499
## DOID:10283    40/331 394/6268 3.859157e-05 0.013921739
## DOID:3459     37/331 357/6268 4.942629e-05 0.013921739
## DOID:3856     40/331 404/6268 6.821467e-05 0.014410349
## DOID:824      16/331 109/6268 1.699304e-04 0.018859464
## DOID:3905     43/331 465/6268 1.749754e-04 0.018859464
##                 qvalue
## DOID:170   0.003826407
## DOID:10283 0.011133923
## DOID:3459  0.011133923
## DOID:3856  0.011524689
## DOID:824   0.015082872
## DOID:3905  0.015082872
##                                                                                                                                                                                                                                                   geneID
## DOID:170   10874/7153/1381/6241/11065/10232/332/6286/2146/10112/891/9232/4171/993/5347/4318/3576/1515/4821/8836/3159/7980/5888/333/898/9768/4288/3551/2152/9590/185/7043/3357/2952/5327/3667/1634/1287/4582/7122/3479/4680/6424/80310/652/8839/9547/1524
## DOID:10283                                          4312/6280/6279/597/3627/332/6286/2146/4321/4521/891/5347/4102/4318/701/3576/79852/10321/6352/4288/3551/2152/247/2952/3487/367/3667/4128/4582/563/3679/4117/7031/3479/6424/10451/80310/652/4036/10551
## DOID:3459                                                          4312/6280/6279/7153/4751/890/4085/332/6286/6790/891/9232/10855/4171/5347/4318/701/2633/3576/9636/898/8792/4288/2952/4982/4128/4582/7031/3479/771/4250/2066/3169/10647/5304/5241/10551
## DOID:3856                                           4312/6280/6279/597/3627/332/6286/2146/4321/4521/891/5347/4102/4318/701/3576/79852/10321/6352/4288/3551/2152/247/2952/3487/367/3667/4128/4582/563/3679/4117/7031/3479/6424/10451/80310/652/4036/10551
## DOID:824                                                                                                                                                                   4312/6279/820/7850/4321/3595/4318/4069/3576/1493/6352/8842/185/2952/5327/4982
## DOID:3905                          4312/6280/2305/9133/6279/7153/6278/6241/55165/11065/8140/10232/332/6286/3002/9212/4521/891/4171/9928/8061/4318/3576/1978/1894/7980/7083/898/6352/8842/4288/2152/2697/2952/3572/4582/7049/563/3479/1846/3117/2532/2922
##            Count
## DOID:170      48
## DOID:10283    40
## DOID:3459     37
## DOID:3856     40
## DOID:824      16
## DOID:3905     43</code></pre>
<p>The <code>enrichDO</code> function requires an entrezgene ID vector as input, mostly is the differential gene list of gene expression profile studies. If user needs to convert other gene ID type to entrezgene ID, we recommend using <code>bitr</code> function provided by <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a>.</p>
<p>The <code>ont</code> parameter can be “DO” or “DOLite”, DOLite<span class="citation">(Du et al. <a href="#ref-Du15062009">2009</a>)</span> was constructed to aggregate the redundant DO terms. The DOLite data is not updated, we recommend user use <code>ont=&quot;DO&quot;</code>. <code>pvalueCutoff</code> setting the cutoff value of <em>p</em> value and <em>p</em> value adjust; <code>pAdjustMethod</code> setting the <em>p</em> value correction methods, include the Bonferroni correction (“bonferroni”), Holm (“holm”), Hochberg (“hochberg”), Hommel (“hommel”), Benjamini &amp; Hochberg (“BH”) and Benjamini &amp; Yekutieli (“BY”) while <code>qvalueCutoff</code> is used to control <em>q</em>-values.</p>
<p>The <code>universe</code> setting the background gene universe for testing. If user do not explicitly setting this parameter, <code>enrichDO</code> will set the universe to all human genes that have DO annotation.</p>
<p>The <code>minGSSize</code> (and <code>maxGSSize</code>) indicates that only those DO terms that have more than <code>minGSSize</code> (and less than <code>maxGSSize</code>) genes annotated will be tested.</p>
<p>The <code>readable</code> is a logical parameter, indicates whether the entrezgene IDs will mapping to gene symbols or not.</p>
<p>We also implement <code>setReadable</code> function that helps the user to convert entrezgene IDs to gene symbols.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">setReadable</span>(x, <span class="st">&#39;org.Hs.eg.db&#39;</span>)
<span class="kw">head</span>(x)</code></pre></div>
<pre><code>##                    ID                    Description
## DOID:170     DOID:170         endocrine gland cancer
## DOID:10283 DOID:10283                prostate cancer
## DOID:3459   DOID:3459               breast carcinoma
## DOID:3856   DOID:3856 male reproductive organ cancer
## DOID:824     DOID:824                  periodontitis
## DOID:3905   DOID:3905                 lung carcinoma
##            GeneRatio  BgRatio       pvalue    p.adjust
## DOID:170      48/331 472/6268 5.662129e-06 0.004784499
## DOID:10283    40/331 394/6268 3.859157e-05 0.013921739
## DOID:3459     37/331 357/6268 4.942629e-05 0.013921739
## DOID:3856     40/331 404/6268 6.821467e-05 0.014410349
## DOID:824      16/331 109/6268 1.699304e-04 0.018859464
## DOID:3905     43/331 465/6268 1.749754e-04 0.018859464
##                 qvalue
## DOID:170   0.003826407
## DOID:10283 0.011133923
## DOID:3459  0.011133923
## DOID:3856  0.011524689
## DOID:824   0.015082872
## DOID:3905  0.015082872
##                                                                                                                                                                                                                                                                                         geneID
## DOID:170   NMU/TOP2A/CRABP1/RRM2/UBE2C/MSLN/BIRC5/S100P/EZH2/KIF20A/CCNB1/PTTG1/MCM2/CDC25A/PLK1/MMP9/CXCL8/CTSV/NKX2-2/GGH/HMGA1/TFPI2/RAD51/APLP1/CCNE1/PCLAF/MKI67/IKBKB/F3/AKAP12/AGTR1/TGFB3/HTR2B/GSTT1/PLAT/IRS1/DCN/COL4A5/MUC1/CLDN5/IGF1/CEACAM6/SFRP4/PDGFD/BMP4/CCN5/CXCL14/CX3CR1
## DOID:10283                                                  MMP1/S100A9/S100A8/BCL2A1/CXCL10/BIRC5/S100P/EZH2/MMP12/NUDT1/CCNB1/PLK1/MAGEA3/MMP9/BUB1B/CXCL8/EPHX3/CRISP3/CCL5/MKI67/IKBKB/F3/ALOX15B/GSTT1/IGFBP4/AR/IRS1/MAOA/MUC1/AZGP1/ITGA7/MAK/TFF1/IGF1/SFRP4/VAV3/PDGFD/BMP4/LRP2/AGR2
## DOID:3459                                                          MMP1/S100A9/S100A8/TOP2A/NEK2/CCNA2/MAD2L1/BIRC5/S100P/AURKA/CCNB1/PTTG1/HPSE/MCM2/PLK1/MMP9/BUB1B/GBP1/CXCL8/ISG15/CCNE1/TNFRSF11A/MKI67/GSTT1/TNFRSF11B/MAOA/MUC1/TFF1/IGF1/CA12/SCGB2A2/ERBB4/FOXA1/SCGB1D2/PIP/PGR/AGR2
## DOID:3856                                                   MMP1/S100A9/S100A8/BCL2A1/CXCL10/BIRC5/S100P/EZH2/MMP12/NUDT1/CCNB1/PLK1/MAGEA3/MMP9/BUB1B/CXCL8/EPHX3/CRISP3/CCL5/MKI67/IKBKB/F3/ALOX15B/GSTT1/IGFBP4/AR/IRS1/MAOA/MUC1/AZGP1/ITGA7/MAK/TFF1/IGF1/SFRP4/VAV3/PDGFD/BMP4/LRP2/AGR2
## DOID:824                                                                                                                                                                                       MMP1/S100A8/CAMP/IL1R2/MMP12/IL12RB2/MMP9/LYZ/CXCL8/CTLA4/CCL5/PROM1/AGTR1/GSTT1/PLAT/TNFRSF11B
## DOID:3905                           MMP1/S100A9/FOXM1/CCNB2/S100A8/TOP2A/S100A7/RRM2/CEP55/UBE2C/SLC7A5/MSLN/BIRC5/S100P/GZMB/AURKB/NUDT1/CCNB1/MCM2/KIF14/FOSL1/MMP9/CXCL8/EIF4EBP1/ECT2/TFPI2/TK1/CCNE1/CCL5/PROM1/MKI67/F3/GJA1/GSTT1/IL6ST/MUC1/TGFBR3/AZGP1/IGF1/DUSP4/HLA-DQA1/ACKR1/GRP
##            Count
## DOID:170      48
## DOID:10283    40
## DOID:3459     37
## DOID:3856     40
## DOID:824      16
## DOID:3905     43</code></pre>
</div>
<div id="enrichncg-function" class="section level2">
<h2><span class="header-section-number">4.2</span> <code>enrichNCG</code> function</h2>
<p><a href="http://ncg.kcl.ac.uk/">Network of Cancer Gene</a> (NCG)<span class="citation">(A. et al. <a href="#ref-omer_ncg">2016</a>)</span> is a manually curated repository of cancer genes. NCG release 5.0 (Aug. 2015) collects 1,571 cancer genes from 175 published studies. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> supports analyzing gene list and determine whether they are enriched in genes known to be mutated in a given cancer type.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene2 &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&lt;</span><span class="st"> </span><span class="dv">3</span>]
ncg &lt;-<span class="st"> </span><span class="kw">enrichNCG</span>(gene2)
<span class="kw">head</span>(ncg)</code></pre></div>
<pre><code>##                                        ID
## soft_tissue_sarcomas soft_tissue_sarcomas
## bladder                           bladder
## glioma                             glioma
##                               Description GeneRatio BgRatio
## soft_tissue_sarcomas soft_tissue_sarcomas   28/1172 28/1571
## bladder                           bladder   61/1172 67/1571
## glioma                             glioma   68/1172 76/1571
##                            pvalue    p.adjust      qvalue
## soft_tissue_sarcomas 0.0002517511 0.008056037 0.006360029
## bladder              0.0005108168 0.008173069 0.006452423
## glioma               0.0008511747 0.009079196 0.007167787
##                                                                                                                                                                                                                                                                                                                                                                  geneID
## soft_tissue_sarcomas                                                                                                                                                                                                       1029/999/6850/4914/4342/2185/55294/2041/4851/23512/2044/4058/5290/8726/4486/5297/5728/3815/2324/7403/5925/4763/1499/7157/5159/2045/3667/2066
## bladder                                            9700/2175/9603/1029/8997/688/1026/896/677/6256/55294/8085/4851/3265/1999/3845/8243/10605/8295/4854/5290/2033/4780/23224/23217/2064/23385/55252/10735/4853/387/288/30849/9794/7403/287/463/472/4297/2065/2262/8289/9611/5925/2068/4763/7157/2186/1387/3910/2261/7248/23037/23345/7832/79633/10628/22906/388/4036/3169
## glioma               4603/4609/1029/3418/8877/1019/7027/4613/1030/1956/1106/2264/3417/6597/4914/55359/896/894/2321/3954/5335/5781/8439/673/9444/4851/8087/2050/8493/3845/3482/667/56999/5290/2033/4233/577/5894/5156/80036/9407/3020/1021/5598/5728/8621/1828/63035/23592/8880/2260/54880/4916/2263/1639/90/546/8289/4763/7157/23152/5295/4602/595/2261/6938/4915/26137
##                      Count
## soft_tissue_sarcomas    28
## bladder                 61
## glioma                  68</code></pre>
</div>
<div id="enrichdgn-and-enrichdgnv-functions" class="section level2">
<h2><span class="header-section-number">4.3</span> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</h2>
<p><a href="http://disgenet.org/">DisGeNET</a><span class="citation">(Janet et al. <a href="#ref-janet_disgenet">2015</a>)</span> is an integrative and comprehensive resources of gene-disease associations from several public data sources and the literature. It contains gene-disease associations and snp-gene-disease associations.</p>
<p>The enrichment analysis of disease-gene associations is supported by the <code>enrichDGN</code> function and analysis of snp-gene-disease associations is supported by the <code>enrichDGNv</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dgn &lt;-<span class="st"> </span><span class="kw">enrichDGN</span>(gene)
<span class="kw">head</span>(dgn)</code></pre></div>
<pre><code>##                          ID
## umls:C1134719 umls:C1134719
## umls:C0032460 umls:C0032460
## umls:C0206698 umls:C0206698
## umls:C0007138 umls:C0007138
## umls:C0031099 umls:C0031099
## umls:C0005695 umls:C0005695
##                                    Description GeneRatio
## umls:C1134719 Invasive Ductal Breast Carcinoma    28/476
## umls:C0032460        Polycystic Ovary Syndrome    38/476
## umls:C0206698               Cholangiocarcinoma    36/476
## umls:C0007138     Carcinoma, Transitional Cell    35/476
## umls:C0031099                    Periodontitis    28/476
## umls:C0005695                 Bladder Neoplasm    36/476
##                 BgRatio       pvalue     p.adjust
## umls:C1134719 231/17381 4.312190e-11 1.225524e-07
## umls:C0032460 434/17381 2.819624e-10 3.521620e-07
## umls:C0206698 399/17381 3.717403e-10 3.521620e-07
## umls:C0007138 389/17381 7.093837e-10 5.040171e-07
## umls:C0031099 270/17381 1.634417e-09 9.290027e-07
## umls:C0005695 442/17381 5.871618e-09 2.781190e-06
##                     qvalue
## umls:C1134719 9.164539e-08
## umls:C0032460 2.633487e-07
## umls:C0206698 2.633487e-07
## umls:C0007138 3.769068e-07
## umls:C0031099 6.947133e-07
## umls:C0005695 2.079789e-06
##                                                                                                                                                                                                         geneID
## umls:C1134719                                                 9133/7153/6241/55165/11065/51203/22974/4751/5080/332/2568/3902/6790/891/24137/9232/10855/79801/4318/55635/5888/1493/9768/3070/4288/367/4582/5241
## umls:C0032460 4312/6280/6279/7153/259266/6241/55165/55872/4085/6286/7272/366/891/4171/7941/1164/3161/4603/990/29127/4318/53335/3294/3070/2952/5327/367/3667/4582/563/27324/3479/114899/9370/2167/652/5346/5241
## umls:C0206698             4312/2305/55872/4751/8140/10635/10232/5918/332/6286/2146/4521/891/10855/2921/7941/1164/4318/3576/1978/79852/8842/4485/214/65982/6863/1036/6935/4128/3572/4582/7031/7166/4680/80310/9
## umls:C0007138                       4312/991/6280/6241/55165/10460/6373/8140/890/10232/4085/332/6286/2146/4171/1033/6364/5347/4318/3576/8836/9700/898/4288/2952/367/8382/2947/3479/9338/23158/2167/2066/2625/9
## umls:C0031099                                                       4312/6279/3669/820/7850/332/4321/6364/3595/4318/3576/3898/8792/1493/4485/10472/185/6863/2205/2952/5327/4982/23261/2200/3572/2006/1308/2625
## umls:C0005695                   4312/10874/6280/3868/6279/597/7153/6241/9582/10460/4085/5080/332/2146/6790/10855/4171/5347/4318/3576/8836/9636/9700/898/4288/214/2952/367/2947/4582/3479/6424/9338/2066/1580/9
##               Count
## umls:C1134719    28
## umls:C0032460    38
## umls:C0206698    36
## umls:C0007138    35
## umls:C0031099    28
## umls:C0005695    36</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">snp &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;rs1401296&quot;</span>, <span class="st">&quot;rs9315050&quot;</span>, <span class="st">&quot;rs5498&quot;</span>, <span class="st">&quot;rs1524668&quot;</span>, <span class="st">&quot;rs147377392&quot;</span>,
         <span class="st">&quot;rs841&quot;</span>, <span class="st">&quot;rs909253&quot;</span>, <span class="st">&quot;rs7193343&quot;</span>, <span class="st">&quot;rs3918232&quot;</span>, <span class="st">&quot;rs3760396&quot;</span>,
         <span class="st">&quot;rs2231137&quot;</span>, <span class="st">&quot;rs10947803&quot;</span>, <span class="st">&quot;rs17222919&quot;</span>, <span class="st">&quot;rs386602276&quot;</span>, <span class="st">&quot;rs11053646&quot;</span>,
         <span class="st">&quot;rs1805192&quot;</span>, <span class="st">&quot;rs139564723&quot;</span>, <span class="st">&quot;rs2230806&quot;</span>, <span class="st">&quot;rs20417&quot;</span>, <span class="st">&quot;rs966221&quot;</span>)
dgnv &lt;-<span class="st"> </span><span class="kw">enrichDGNv</span>(snp)
<span class="kw">head</span>(dgnv)</code></pre></div>
<pre><code>##                          ID
## umls:C3272363 umls:C3272363
## umls:C0948008 umls:C0948008
## umls:C0038454 umls:C0038454
## umls:C0027051 umls:C0027051
## umls:C0010054 umls:C0010054
## umls:C0010068 umls:C0010068
##                                     Description GeneRatio
## umls:C3272363 Ischemic Cerebrovascular Accident     20/20
## umls:C0948008                   Ischemic stroke     20/20
## umls:C0038454          Cerebrovascular accident      7/20
## umls:C0027051             Myocardial Infarction      6/20
## umls:C0010054         Coronary Arteriosclerosis      6/20
## umls:C0010068            Coronary heart disease      6/20
##                 BgRatio       pvalue     p.adjust
## umls:C3272363 141/46589 1.014503e-51 1.379725e-49
## umls:C0948008 148/46589 2.867870e-51 1.950151e-49
## umls:C0038454 243/46589 7.045680e-12 3.194042e-10
## umls:C0027051 163/46589 6.222154e-11 1.889883e-09
## umls:C0010054 166/46589 6.948100e-11 1.889883e-09
## umls:C0010068 314/46589 3.198889e-09 7.250815e-08
##                     qvalue
## umls:C3272363 1.922217e-50
## umls:C0948008 2.716929e-50
## umls:C0038454 4.449903e-11
## umls:C0027051 2.632964e-10
## umls:C0010054 2.632964e-10
## umls:C0010068 1.010175e-08
##                                                                                                                                                                                                              geneID
## umls:C3272363 rs1401296/rs9315050/rs5498/rs1524668/rs147377392/rs841/rs909253/rs7193343/rs3918232/rs3760396/rs2231137/rs10947803/rs17222919/rs386602276/rs11053646/rs1805192/rs139564723/rs2230806/rs20417/rs966221
## umls:C0948008 rs1401296/rs9315050/rs5498/rs1524668/rs147377392/rs841/rs909253/rs7193343/rs3918232/rs3760396/rs2231137/rs10947803/rs17222919/rs386602276/rs11053646/rs1805192/rs139564723/rs2230806/rs20417/rs966221
## umls:C0038454                                                                                                                              rs1524668/rs147377392/rs2231137/rs10947803/rs386602276/rs2230806/rs20417
## umls:C0027051                                                                                                                                              rs5498/rs147377392/rs909253/rs11053646/rs1805192/rs20417
## umls:C0010054                                                                                                                                             rs5498/rs147377392/rs11053646/rs1805192/rs2230806/rs20417
## umls:C0010068                                                                                                                                             rs5498/rs147377392/rs11053646/rs1805192/rs2230806/rs20417
##               Count
## umls:C3272363    20
## umls:C0948008    20
## umls:C0038454     7
## umls:C0027051     6
## umls:C0010054     6
## umls:C0010068     6</code></pre>
</div>
<div id="gsedo-fuction" class="section level2">
<h2><span class="header-section-number">4.4</span> <code>gseDO</code> fuction</h2>
<p>In the following example, in order to speedup the compilation of this document, only gene sets with size above 120 were tested and only 100 permutations were performed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DOSE)
<span class="kw">data</span>(geneList)
y &lt;-<span class="st"> </span><span class="kw">gseDO</span>(geneList,
           <span class="dt">nPerm         =</span> <span class="dv">100</span>,
           <span class="dt">minGSSize     =</span> <span class="dv">120</span>,
           <span class="dt">pvalueCutoff  =</span> <span class="fl">0.2</span>,
           <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,
           <span class="dt">verbose       =</span> <span class="ot">FALSE</span>)
<span class="kw">head</span>(y, <span class="dv">3</span>)</code></pre></div>
<pre><code>##                  ID          Description setSize
## DOID:5679 DOID:5679      retinal disease     299
## DOID:654   DOID:654        overnutrition     298
## DOID:8466 DOID:8466 retinal degeneration     216
##           enrichmentScore       NES     pvalue   p.adjust
## DOID:5679      -0.3676313 -1.555515 0.01219512 0.08038914
## DOID:654       -0.3490198 -1.477495 0.01219512 0.08038914
## DOID:8466      -0.3467372 -1.465667 0.01219512 0.08038914
##              qvalues rank                   leading_edge
## DOID:5679 0.03449484 1768 tags=24%, list=14%, signal=21%
## DOID:654  0.03449484 1464 tags=21%, list=12%, signal=19%
## DOID:8466 0.03449484 1465 tags=21%, list=12%, signal=19%
##                                                                                                                                                                                                                                                                                                                                                                core_enrichment
## DOID:5679 2878/3791/23247/80184/6750/7450/596/9187/2034/482/948/1490/1280/5737/4314/4881/3426/187/629/6403/6785/2934/5176/7078/5950/727/10516/4311/2247/1295/358/10203/582/10218/57125/585/1675/6310/2202/4313/2944/4254/3075/2099/3480/4653/6387/1471/857/4016/1909/4053/6678/1296/4915/55812/1191/5654/10631/2697/2952/6935/2200/3479/2006/10451/9370/771/652/4693/5346/1524
## DOID:654                                                   2169/1490/7840/4887/4314/595/4018/6403/590/3087/866/66036/5919/5176/3953/5950/2638/2166/5243/5468/4023/3485/7350/3952/1149/585/1513/3489/79068/4671/477/4313/3625/9369/6720/7494/2099/3480/3991/6678/4915/5167/8228/165/2152/185/367/4982/3667/4128/9607/3572/150/563/1489/3479/9370/9122/5105/2167/5346/79689/5241
## DOID:8466                                                                                                                                               948/5737/4314/4881/3426/629/6403/6785/2934/5176/7078/727/10516/1295/358/10203/582/10218/585/1675/2202/2944/3075/2099/4653/1471/857/4016/1909/4053/6678/1296/4915/1191/5654/2697/2952/6935/2200/2006/10451/771/652/1524</code></pre>
</div>
<div id="gsencg-fuction" class="section level2">
<h2><span class="header-section-number">4.5</span> <code>gseNCG</code> fuction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncg &lt;-<span class="st"> </span><span class="kw">gseNCG</span>(geneList,
              <span class="dt">nPerm         =</span> <span class="dv">100</span>,
              <span class="dt">minGSSize     =</span> <span class="dv">120</span>,
              <span class="dt">pvalueCutoff  =</span> <span class="fl">0.2</span>,
              <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,
              <span class="dt">verbose       =</span> <span class="ot">FALSE</span>)
ncg &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ncg, <span class="st">&#39;org.Hs.eg.db&#39;</span>)
<span class="kw">head</span>(ncg, <span class="dv">3</span>)</code></pre></div>
<pre><code>##                ID Description setSize enrichmentScore
## breast     breast      breast     133      -0.4869070
## lung         lung        lung     173      -0.3880662
## lymphoma lymphoma    lymphoma     188       0.2999589
##                NES     pvalue   p.adjust    qvalues rank
## breast   -1.893048 0.01369863 0.04166667 0.02192982 2930
## lung     -1.541319 0.01388889 0.04166667 0.02192982 2775
## lymphoma  1.388834 0.03703704 0.07407407 0.03898635 2087
##                            leading_edge
## breast   tags=33%, list=23%, signal=26%
## lung     tags=31%, list=22%, signal=25%
## lymphoma tags=21%, list=17%, signal=18%
##                                                                                                                                                                                                         core_enrichment
## breast                                                                                   KMT2A/ERBB3/SETD2/ARID1A/GPS2/NCOR1/RB1/MAP2K4/NF1/TP53/PIK3R1/STK11/CDKN1B/PTGFR/APC/CCND1/TRAF5/MAP3K1/ESR1/TBX3/FOXA1/GATA3
## lung     SETD2/ATXN3L/LRP1B/BRD3/ARID1A/INHBA/RB1/ADCY1/LYRM9/NF1/CTNNB1/TP53/SATB2/STK11/CTIF/CTNNA3/KDR/COL11A1/FLT3/APC/ADGRL3/FGFR3/NCAM2/DIP2C/APLNR/SLIT2/EPHA3/RUNX1T1/ZMYND10/ZFHX4/GLI3/TNN/PLSCR4/DACH1/ERBB4
## lymphoma                                        DUSP2/EZH2/PRDM1/MYC/ZWILCH/IKZF3/PLCG2/IDH2/HIST1H1C/MAGEC3/CD79B/ETV6/HIST1H1E/HIST1H1B/IRF8/CD28/SLC29A2/DUSP9/TNFAIP3/DNMT3A/SYK/TNF/BCR/HIST1H1D/DSC3/UBE2A/PABPC1</code></pre>
</div>
<div id="gsedgn-fuction" class="section level2">
<h2><span class="header-section-number">4.6</span> <code>gseDGN</code> fuction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dgn &lt;-<span class="st"> </span><span class="kw">gseDGN</span>(geneList,
              <span class="dt">nPerm         =</span> <span class="dv">100</span>,
              <span class="dt">minGSSize     =</span> <span class="dv">120</span>,
              <span class="dt">pvalueCutoff  =</span> <span class="fl">0.2</span>,
              <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,
              <span class="dt">verbose       =</span> <span class="ot">FALSE</span>)
dgn &lt;-<span class="st"> </span><span class="kw">setReadable</span>(dgn, <span class="st">&#39;org.Hs.eg.db&#39;</span>)
<span class="kw">head</span>(dgn, <span class="dv">3</span>)</code></pre></div>
<pre><code>##                          ID         Description setSize
## umls:C0011581 umls:C0011581 Depressive disorder     464
## umls:C0011570 umls:C0011570   Mental Depression     483
## umls:C0151744 umls:C0151744 Myocardial Ischemia     418
##               enrichmentScore       NES     pvalue
## umls:C0011581      -0.2963136 -1.325750 0.01149425
## umls:C0011570      -0.2874181 -1.296230 0.01162791
## umls:C0151744      -0.3013524 -1.349414 0.01162791
##                p.adjust   qvalues rank
## umls:C0011581 0.1143293 0.0724754 2587
## umls:C0011570 0.1143293 0.0724754 2587
## umls:C0151744 0.1143293 0.0724754 2309
##                                 leading_edge
## umls:C0011581 tags=29%, list=21%, signal=24%
## umls:C0011570 tags=25%, list=21%, signal=20%
## umls:C0151744 tags=26%, list=18%, signal=22%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            core_enrichment
## umls:C0011581 ATM/TRH/CYP2E1/IFNAR1/TTR/CALD1/BCL2L1/CRH/SUCLA2/CNR1/MTR/ARNTL/PER1/GDE1/SLC29A3/FOSB/TH/HTR7/SMPD1/ETS2/HDAC5/RGN/GRIA1/PTGS1/PDE4A/SNCA/ADAMTS2/EHD3/NR5A1/SORCS3/CRY1/ADRB2/FZD1/MYOM2/ADCY1/POU6F1/MAPK3/BICC1/SLC6A4/AHI1/TP53/RNF103/SLC12A2/BDNF/NR3C1/SRSF5/PCLO/GABRA6/WWC1/IL5/GLUL/ELK3/GAD1/RARA/GRM5/KDR/ASAH1/IMPACT/CHRM2/WFS1/TSPAN31/HP/PVALB/HTR1A/BCL2/GPM6A/CYP2A6/DUSP1/NLGN4Y/F2R/CD36/NGFR/NPY2R/DBH/BECN1/CCND1/OXTR/SGCE/SELP/NGF/LPAR1/NRP1/AVPR1B/IFT88/ARSD/FAAH/NEFL/FGF2/CD1C/ABCB1/SRPX/RAPGEF3/CRHBP/HSPA2/LEP/FTO/PER2/ALPK1/GSTM1/DIXDC1/XBP1/ESR1/IGF1R/NTF3/CACNA1C/NR3C2/SLC18A2/NTRK2/SPDEF/RAPGEF4/ALB/NPY1R/F3/AGTR1/TAC1/AR/UCN/FBN1/MAOA/CARTPT/TAT/ADRA2A/MUC1/TGFBR3/TPH1/IGF1/ABAT/MAOB/ADIPOQ/TBC1D9/ADH1B/CRY2/GATA3/TFAP2B
## umls:C0011570                                                                                 ETS2/RGN/GRIA1/PTGS1/NLGN1/PDE4A/ADAMTS2/EHD3/NR5A1/SORCS3/A2M/KCNQ1/CRY1/ADRB2/FZD1/MYOM2/ADCY1/POU6F1/MAPK3/BICC1/SLC6A4/AHI1/TP53/DBP/SLC12A2/BDNF/NR3C1/SRSF5/PCLO/GABRA6/WWC1/IL5/GLUL/ELK3/GAD1/RARA/GRM5/ASAH1/IMPACT/CHRM2/WFS1/TSPAN31/ARGLU1/HP/PVALB/HTR1A/GPM6A/CYP2A6/DUSP1/NLGN4Y/F2R/CD36/DBH/BECN1/CCND1/PER3/OXTR/SGCE/CFB/CLASP2/LPAR1/NRP1/AVPR1B/ARSD/GC/FAAH/BHLHE41/FGF2/CD1C/ABCB1/PPARG/SRPX/RAPGEF3/CRHBP/CDH13/HSPA2/BHLHE40/PDE1A/LEP/FTO/PER2/ALPK1/GSTM1/DIXDC1/XBP1/TCF4/ESR1/IGF1R/NTF3/CACNA1C/NR3C2/SLC18A2/NTRK2/RAPGEF4/F3/AGTR1/TAC1/GSTT1/AR/UCN/FBN1/MAOA/CARTPT/TAT/ADRA2A/MUC1/TGFBR3/TPH1/IGF1/MAOB/ADIPOQ/TBC1D9/ADH1B/EMX2/MAPT/CRY2/GATA3/TFAP2B
## umls:C0151744                                                                                                                                           ADRB2/HSPB1/ABCC6/ADD1/PECAM1/MAPK3/GRK5/VEGFC/AMPD1/TLE5/F7/HSPB2/ENTPD1/ID3/PRKAA2/ATP2B1/SOD3/PRKAB1/AMH/STAT6/RGCC/RXRG/GDF10/SLC9A1/HGF/SERPINA3/MBL2/KDR/EGR1/HSPB6/HBB/STAT5A/EEF1A2/VWF/BCL2/CD34/DUSP1/PRKG1/CD36/CCN2/MMP3/BECN1/NPR1/CCND1/GATM/LPA/EDIL3/RTN1/APLNR/PYGM/SELP/FGF1/NEDD4/ID1/ALDH6A1/FOXO1/SULT1A1/SNAP23/FGF2/DUSP6/ABCB1/PPARG/PDK4/SHH/HSPA2/BHLHE40/LPL/THBD/COL5A2/UGCG/KL/ADH1C/GSTM2/THBS2/PER2/ATXN1/MMP2/TXNIP/KITLG/CFH/ESR1/CXCL12/CIRBP/EDNRA/GHR/SPARC/GPD1L/ENPP1/ALB/F13A1/MEOX2/F3/AGTR1/ZEB1/TNFRSF11B/UCN/DCN/LTC4S/IL6ST/EPHX2/THBS4/IGF1/FXYD1/SFRP4/ELN/RAMP2/ADIPOQ/ADH1B/HMGCS2</code></pre>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-omer_ncg">
<p>A., Omer, Giovanni M. D., Thanos P. M., and Francesca D. C. 2016. “NCG 5.0: Updates of a Manually Curated Repository of Cancer Genes and Associated Properties from Cancer Mutational Screenings.” <em>Nucleic Acids Research</em> 44 (D1): D992–D999. doi:<a href="https://doi.org/10.1093/nar/gkv1123">10.1093/nar/gkv1123</a>.</p>
</div>
<div id="ref-Du15062009">
<p>Du, Pan, Gang Feng, Jared Flatow, Jie Song, Michelle Holko, Warren A. Kibbe, and Simon M. Lin. 2009. “From Disease Ontology to Disease-Ontology Lite: Statistical Methods to Adapt a General-Purpose Ontology for the Test of Gene-Ontology Associations.” <em>Bioinformatics</em> 25 (12): i63–i68. doi:<a href="https://doi.org/10.1093/bioinformatics/btp193">10.1093/bioinformatics/btp193</a>.</p>
</div>
<div id="ref-janet_disgenet">
<p>Janet, P., Núria Q.R., Àlex B., Jordi D.P., Anna B.M., Martin B., Ferran S., and Laura I. F. 2015. “DisGeNET: A Discovery Platform for the Dynamical Exploration of Human Diseases and Their Genes.” <em>Database</em> 2015 (March): bav028. doi:<a href="https://doi.org/10.1093/database/bav028">10.1093/database/bav028</a>.</p>
</div>
<div id="ref-yu_dose_2015">
<p>Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. “DOSE: An R/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.” <em>Bioinformatics</em> 31 (4): 608–9. doi:<a href="https://doi.org/10.1093/bioinformatics/btu684">10.1093/bioinformatics/btu684</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
