<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Universal enrichment analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter2.html">
<link rel="next" href="chapter4.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.6/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#terminology"><i class="fa fa-check"></i><b>1.1</b> Terminology</a><ul>
<li class="chapter" data-level="1.1.1" data-path="chapter1.html"><a href="chapter1.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>1.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="1.1.2" data-path="chapter1.html"><a href="chapter1.html#gene-ontology-go"><i class="fa fa-check"></i><b>1.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="1.1.3" data-path="chapter1.html"><a href="chapter1.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>1.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="1.1.4" data-path="chapter1.html"><a href="chapter1.html#other-gene-sets"><i class="fa fa-check"></i><b>1.1.4</b> Other gene sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Functional Enrichment Analysis Methods</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#over-representation-analysis"><i class="fa fa-check"></i><b>2.1</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.3</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#input-data"><i class="fa fa-check"></i><b>3.1</b> Input data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#cell-marker"><i class="fa fa-check"></i><b>3.3</b> Cell Marker</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#msigdb-analysis"><i class="fa fa-check"></i><b>3.4</b> MSigDb analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="chapter4.html"><a href="chapter4.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Gene Ontology Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter5.html"><a href="chapter5.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="chapter5.html"><a href="chapter5.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chapter5.html"><a href="chapter5.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>5.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="chapter12.html"><a href="chapter12.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="chapter12.html"><a href="chapter12.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="chapter12.html"><a href="chapter12.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="chapter12.html"><a href="chapter12.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="chapter12.html"><a href="chapter12.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> dplyr verbs for clusterProfiler</a><ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#filter"><i class="fa fa-check"></i><b>13.1</b> filter</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#arrange"><i class="fa fa-check"></i><b>13.2</b> arrange</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#select"><i class="fa fa-check"></i><b>13.3</b> select</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#mutate"><i class="fa fa-check"></i><b>13.4</b> mutate</a></li>
<li class="chapter" data-level="13.5" data-path="chapter13.html"><a href="chapter13.html#summarise"><i class="fa fa-check"></i><b>13.5</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="chapter14.html"><a href="chapter14.html"><i class="fa fa-check"></i><b>14</b> Useful utilities</a><ul>
<li class="chapter" data-level="14.1" data-path="chapter14.html"><a href="chapter14.html#bitr"><i class="fa fa-check"></i><b>14.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="14.1.1" data-path="chapter14.html"><a href="chapter14.html#bitr_kegg"><i class="fa fa-check"></i><b>14.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="chapter14.html"><a href="chapter14.html#setReadable"><i class="fa fa-check"></i><b>14.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter3" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Universal enrichment analysis</h1>
<p><a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> supports both hypergeometric test and gene set enrichment analyses of many ontology/pathway, but it’s still not enough for users may want to analyze their data with unsupported organisms, slim version of GO, novel functional annotation (e.g. GO via BlastGO or KEGG via KAAS), unsupported ontologies/pathways or customized annotations.</p>
<p><a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a> provides <code>enricher</code> function for hypergeometric test and <code>GSEA</code> function for gene set enrichment analysis that are designed to accept user defined annotation. They accept two additional parameters <em>TERM2GENE</em> and <em>TERM2NAME</em>. As indicated in the parameter names, <em>TERM2GENE</em> is a data.frame with first column of term ID and second column of corresponding mapped gene and <em>TERM2NAME</em> is a data.frame with first column of term ID and second column of corresponding term name. <em>TERM2NAME</em> is optional.</p>
<div id="input-data" class="section level2">
<h2><span class="header-section-number">3.1</span> Input data</h2>
<p>For over representation analysis, all we need is a gene vector, that is a vector of gene IDs. These gene IDs can be obtained by differential expression analysis (<em>e.g.</em> with <a href="http://www.bioconductor.org/packages/DESeq2">DESeq2</a> package).</p>
<p>For gene set enrichment analysis, we need a ranked list of genes. <a href="http://bioconductor.org/packages/DOSE">DOSE</a> provides an example dataset <code>geneList</code> which was derived from <code>R</code> package <a href="http://bioconductor.org/packages/breastCancerMAINZ">breastCancerMAINZ</a> that contained 200 samples, including 29 samples in grade I, 136 samples in grade II and 35 samples in grade III. We computed the ratios of geometric means of grade III samples versus geometric means of grade I samples. Logarithm of these ratios (base 2) were stored in <code>geneList</code> dataset.</p>
<p>The <code>geneList</code> contains three features:</p>
<ol style="list-style-type: decimal">
<li>numeric vector: fold change or other type of numerical variable</li>
<li>named vector: every number was named by the corresponding gene ID</li>
<li>sorted vector: number should be sorted in decreasing order</li>
</ol>
<p>Suppose you are importing your own data from a <em>csv</em> file and the file contains two columns, one for gene ID (no duplicated allowed) and another one for fold change, you can prepare your own <code>geneList</code> via the following command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">d &lt;-<span class="st"> </span><span class="kw">read.csv</span>(your_csv_file)
## assume that 1st column is ID
## 2nd column is fold change

## feature 1: numeric vector
geneList &lt;-<span class="st"> </span>d[,<span class="dv">2</span>]

## feature 2: named vector
<span class="kw">names</span>(geneList) &lt;-<span class="st"> </span><span class="kw">as.character</span>(d[,<span class="dv">1</span>])

## feature 3: decreasing order
geneList &lt;-<span class="st"> </span><span class="kw">sort</span>(geneList, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>We can load the sample data into R via:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)
<span class="kw">head</span>(geneList)</code></pre></div>
<pre><code>##     4312     8318    10874    55143    55388      991 
## 4.572613 4.514594 4.418218 4.144075 3.876258 3.677857</code></pre>
<p>Suppose we define fold change greater than 2 as DEGs:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]
<span class="kw">head</span>(gene)</code></pre></div>
<pre><code>## [1] &quot;4312&quot;  &quot;8318&quot;  &quot;10874&quot; &quot;55143&quot; &quot;55388&quot; &quot;991&quot;</code></pre>
</div>
<div id="wikipathways-analysis" class="section level2">
<h2><span class="header-section-number">3.2</span> WikiPathways analysis</h2>
<p><a href="https://www.wikipathways.org">WikiPathways</a> is a continuously updated pathway database curated by a community of researchers and pathway enthusiasts. WikiPathways produces monthly releases of gmt files for supported organisms at <a href="http://data.wikipathways.org/current/gmt/">data.wikipathways.org</a>. Download the appropriate gmt file and then generate <code>TERM2GENE</code> and <code>TERM2NAME</code> to use <code>enricher</code> and <code>GSEA</code> functions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(magrittr)
<span class="kw">library</span>(clusterProfiler)

<span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)
gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]

wpgmtfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/wikipathways-20180810-gmt-Homo_sapiens.gmt&quot;</span>, <span class="dt">package=</span><span class="st">&quot;clusterProfiler&quot;</span>)
wp2gene &lt;-<span class="st"> </span><span class="kw">read.gmt</span>(wpgmtfile)
wp2gene &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>tidyr<span class="op">::</span><span class="kw">separate</span>(ont, <span class="kw">c</span>(<span class="st">&quot;name&quot;</span>,<span class="st">&quot;version&quot;</span>,<span class="st">&quot;wpid&quot;</span>,<span class="st">&quot;org&quot;</span>), <span class="st">&quot;%&quot;</span>)
wpid2gene &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(wpid, gene) <span class="co">#TERM2GENE</span>
wpid2name &lt;-<span class="st"> </span>wp2gene <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(wpid, name) <span class="co">#TERM2NAME</span>

ewp &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE =</span> wpid2gene, <span class="dt">TERM2NAME =</span> wpid2name)
<span class="kw">head</span>(ewp)</code></pre></div>
<pre><code>##            ID
## WP2446 WP2446
## WP2361 WP2361
## WP179   WP179
## WP3942 WP3942
## WP4240 WP4240
## WP2328 WP2328
##                                                                           Description
## WP2446                                                  Retinoblastoma (RB) in Cancer
## WP2361                                                       Gastric Cancer Network 1
## WP179                                                                      Cell Cycle
## WP3942                                                         PPAR signaling pathway
## WP4240 Regulation of sister chromatid separation at the metaphase-anaphase transition
## WP2328                                                            Allograft Rejection
##        GeneRatio  BgRatio       pvalue     p.adjust
## WP2446     11/95  88/6249 6.801697e-08 1.054263e-05
## WP2361      6/95  29/6249 3.772735e-06 2.923870e-04
## WP179      10/95 122/6249 1.384549e-05 7.153503e-04
## WP3942      7/95  67/6249 6.210513e-05 2.406574e-03
## WP4240      4/95  16/6249 7.931988e-05 2.458916e-03
## WP2328      7/95  90/6249 4.016758e-04 1.037663e-02
##              qvalue
## WP2446 9.450779e-06
## WP2361 2.621058e-04
## WP179  6.412648e-04
## WP3942 2.157336e-03
## WP4240 2.204258e-03
## WP2328 9.301967e-03
##                                                       geneID
## WP2446 8318/9133/7153/6241/890/983/81620/7272/1111/891/24137
## WP2361                       4605/7153/11065/22974/6286/6790
## WP179          8318/991/9133/890/983/7272/1111/891/4174/9232
## WP3942                    4312/9415/9370/5105/2167/3158/5346
## WP4240                                    991/1062/4085/9232
## WP2328                   10563/6373/4283/3002/10578/3117/730
##        Count
## WP2446    11
## WP2361     6
## WP179     10
## WP3942     7
## WP4240     4
## WP2328     7</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ewp2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> wpid2gene, <span class="dt">TERM2NAME =</span> wpid2name, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="kw">head</span>(ewp2)</code></pre></div>
<pre><code>## [1] ID              Description     setSize        
## [4] enrichmentScore NES             pvalue         
## [7] p.adjust        qvalues        
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>You may want to convert the gene IDs to gene symbols, which can be done by <code>setReadable</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(org.Hs.eg.db)
ewp &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ewp, org.Hs.eg.db, <span class="dt">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>)
ewp2 &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ewp2, org.Hs.eg.db, <span class="dt">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>)
<span class="kw">head</span>(ewp)</code></pre></div>
<pre><code>##            ID
## WP2446 WP2446
## WP2361 WP2361
## WP179   WP179
## WP3942 WP3942
## WP4240 WP4240
## WP2328 WP2328
##                                                                           Description
## WP2446                                                  Retinoblastoma (RB) in Cancer
## WP2361                                                       Gastric Cancer Network 1
## WP179                                                                      Cell Cycle
## WP3942                                                         PPAR signaling pathway
## WP4240 Regulation of sister chromatid separation at the metaphase-anaphase transition
## WP2328                                                            Allograft Rejection
##        GeneRatio  BgRatio       pvalue     p.adjust
## WP2446     11/95  88/6249 6.801697e-08 1.054263e-05
## WP2361      6/95  29/6249 3.772735e-06 2.923870e-04
## WP179      10/95 122/6249 1.384549e-05 7.153503e-04
## WP3942      7/95  67/6249 6.210513e-05 2.406574e-03
## WP4240      4/95  16/6249 7.931988e-05 2.458916e-03
## WP2328      7/95  90/6249 4.016758e-04 1.037663e-02
##              qvalue
## WP2446 9.450779e-06
## WP2361 2.621058e-04
## WP179  6.412648e-04
## WP3942 2.157336e-03
## WP4240 2.204258e-03
## WP2328 9.301967e-03
##                                                              geneID
## WP2446 CDC45/CCNB2/TOP2A/RRM2/CCNA2/CDK1/CDT1/TTK/CHEK1/CCNB1/KIF4A
## WP2361                           MYBL2/TOP2A/UBE2C/TPX2/S100P/AURKA
## WP179       CDC45/CDC20/CCNB2/CCNA2/CDK1/TTK/CHEK1/CCNB1/MCM5/PTTG1
## WP3942                    MMP1/FADS2/ADIPOQ/PCK1/FABP4/HMGCS2/PLIN1
## WP4240                                     CDC20/CENPE/MAD2L1/PTTG1
## WP2328                    CXCL13/CXCL11/CXCL9/GZMB/GNLY/HLA-DQA1/C7
##        Count
## WP2446    11
## WP2361     6
## WP179     10
## WP3942     7
## WP4240     4
## WP2328     7</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(ewp2)</code></pre></div>
<pre><code>## [1] ID              Description     setSize        
## [4] enrichmentScore NES             pvalue         
## [7] p.adjust        qvalues        
## &lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>As an alternative to manually downloading gmt files, install the <a href="https://bioconductor.org/packages/release/bioc/html/rWikiPathways.html">rWikiPathways package</a> to gain scripting access to the latest gmt files using the <code>downloadPathwayArchive</code> function.</p>
</div>
<div id="cell-marker" class="section level2">
<h2><span class="header-section-number">3.3</span> Cell Marker</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell_markers &lt;-<span class="st"> </span>vroom<span class="op">::</span><span class="kw">vroom</span>(<span class="st">&#39;http://bio-bigdata.hrbmu.edu.cn/CellMarker/download/Human_cell_markers.txt&#39;</span>) <span class="op">%&gt;%</span>
<span class="st">   </span>tidyr<span class="op">::</span><span class="kw">unite</span>(<span class="st">&quot;cellMarker&quot;</span>, tissueType, cancerType, cellName, <span class="dt">sep=</span><span class="st">&quot;, &quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">   </span>dplyr<span class="op">::</span><span class="kw">select</span>(cellMarker, geneID)
cell_markers</code></pre></div>
<pre><code>## # A tibble: 2,868 x 2
##    cellMarker                                geneID         
##    &lt;chr&gt;                                     &lt;chr&gt;          
##  1 Kidney, Normal, Proximal tubular cell     248            
##  2 Liver, Normal, Ito cell (hepatic stellat… 6855           
##  3 Endometrium, Normal, Trophoblast cell     634            
##  4 Germ, Normal, Primordial germ cell        54514          
##  5 Corneal epithelium, Normal, Epithelial c… 1316           
##  6 Placenta, Normal, Cytotrophoblast         2255           
##  7 Periosteum, Normal, Periosteum-derived p… 214, 5788, 928…
##  8 Amniotic membrane, Normal, Amnion epithe… 79923, 5460    
##  9 Primitive streak, Normal, Primitive stre… 3975, 83881    
## 10 Adipose tissue, Normal, Stromal vascular… 947            
## # … with 2,858 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">y &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>cell_markers, <span class="dt">minGSSize=</span><span class="dv">1</span>)
DT<span class="op">::</span><span class="kw">datatable</span>(<span class="kw">as.data.frame</span>(y))</code></pre></div>
<div id="htmlwidget-aea1fb48c514cb89f20f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-aea1fb48c514cb89f20f">{"x":{"filter":"none","data":[["Brain, Normal, Lake et al.Science.In8","Eye, Normal, Amacrine cell","Stomach, Gastric Cancer, Oxyntic progenitor cell","Stomach, Gastric Cancer, Oxyntic stem cell","Stomach, Normal, Oxyntic progenitor cell","Stomach, Normal, Oxyntic stem cell","Stomach, Normal, Progenitor cell","Lung, Normal, M1 macrophage","Lung, Normal, M2 macrophage","Undefined, Normal, Endothelial cell","Lung, Normal, Endothelial cell","Undefined, Melanoma, Cancer stem cell"],["Brain, Normal, Lake et al.Science.In8","Eye, Normal, Amacrine cell","Stomach, Gastric Cancer, Oxyntic progenitor cell","Stomach, Gastric Cancer, Oxyntic stem cell","Stomach, Normal, Oxyntic progenitor cell","Stomach, Normal, Oxyntic stem cell","Stomach, Normal, Progenitor cell","Lung, Normal, M1 macrophage","Lung, Normal, M2 macrophage","Undefined, Normal, Endothelial cell","Lung, Normal, Endothelial cell","Undefined, Melanoma, Cancer stem cell"],["Brain, Normal, Lake et al.Science.In8","Eye, Normal, Amacrine cell","Stomach, Gastric Cancer, Oxyntic progenitor cell","Stomach, Gastric Cancer, Oxyntic stem cell","Stomach, Normal, Oxyntic progenitor cell","Stomach, Normal, Oxyntic stem cell","Stomach, Normal, Progenitor cell","Lung, Normal, M1 macrophage","Lung, Normal, M2 macrophage","Undefined, Normal, Endothelial cell","Lung, Normal, Endothelial cell","Undefined, Melanoma, Cancer stem cell"],["1/7","1/7","1/7","1/7","1/7","1/7","1/7","1/7","1/7","1/7","1/7","1/7"],["1/1596","1/1596","1/1596","1/1596","1/1596","1/1596","1/1596","2/1596","2/1596","6/1596","7/1596","7/1596"],[0.00438596491228094,0.00438596491228094,0.00438596491228094,0.00438596491228094,0.00438596491228094,0.00438596491228094,0.00438596491228094,0.00875543089699171,0.00875543089699171,0.0260693386767984,0.030357083726649,0.030357083726649],[0.00751879699248161,0.00751879699248161,0.00751879699248161,0.00751879699248161,0.00751879699248161,0.00751879699248161,0.00751879699248161,0.0116739078626556,0.0116739078626556,0.030357083726649,0.030357083726649,0.030357083726649],[null,null,null,null,null,null,null,null,null,null,null,null],["10874","1381","259266","259266","259266","259266","259266","3627","6362","51705","51705","2146"],[1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>Description<\/th>\n      <th>GeneRatio<\/th>\n      <th>BgRatio<\/th>\n      <th>pvalue<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[5,6,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="msigdb-analysis" class="section level2">
<h2><span class="header-section-number">3.4</span> MSigDb analysis</h2>
<p><a href="http://software.broadinstitute.org/gsea/msigdb">Molecular Signatures Database</a> contains 8 major collections:</p>
<ul>
<li>H: hallmark gene sets</li>
<li>C1: positional gene sets</li>
<li>C2: curated gene sets</li>
<li>C3: motif gene sets</li>
<li>C4: computational gene sets</li>
<li>C5: GO gene sets</li>
<li>C6: oncogenic signatures</li>
<li>C7: immunologic signatures</li>
</ul>
<p>Users can download GMT files from <a href="http://software.broadinstitute.org/gsea/msigdb">Broad Institute</a> and use <code>read.gmt</code> to parse the file to be used in <code>enricher()</code> and <code>GSEA()</code>.</p>
<p>There is an R package, <a href="https://cran.r-project.org/package=msigdbr">msigdbr</a>, that already packed the MSigDB gene sets in tidy data format that can be used directly with <em>clusterProfiler</em>.</p>
<p>It supports several specices:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(msigdbr)
<span class="kw">msigdbr_show_species</span>()</code></pre></div>
<pre><code>##  [1] &quot;Bos taurus&quot;               &quot;Caenorhabditis elegans&quot;  
##  [3] &quot;Canis lupus familiaris&quot;   &quot;Danio rerio&quot;             
##  [5] &quot;Drosophila melanogaster&quot;  &quot;Gallus gallus&quot;           
##  [7] &quot;Homo sapiens&quot;             &quot;Mus musculus&quot;            
##  [9] &quot;Rattus norvegicus&quot;        &quot;Saccharomyces cerevisiae&quot;
## [11] &quot;Sus scrofa&quot;</code></pre>
<p>We can retrieve all human gene sets:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m_df &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>)
<span class="kw">head</span>(m_df, <span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span>as.data.frame</code></pre></div>
<pre><code>##          gs_name  gs_id gs_cat gs_subcat human_gene_symbol
## 1 AAACCAC_MIR140 M12609     C3       MIR             ABCC4
## 2 AAACCAC_MIR140 M12609     C3       MIR             ACTN4
##   species_name entrez_gene gene_symbol sources
## 1 Homo sapiens       10257       ABCC4    &lt;NA&gt;
## 2 Homo sapiens          81       ACTN4    &lt;NA&gt;</code></pre>
<p>Or specific collection. Here we use C6, oncogenic gene sets as an example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C6&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)
<span class="kw">head</span>(m_t2g)</code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name              entrez_gene
##   &lt;chr&gt;                      &lt;int&gt;
## 1 AKT_UP_MTOR_DN.V1_DN       25864
## 2 AKT_UP_MTOR_DN.V1_DN          95
## 3 AKT_UP_MTOR_DN.V1_DN      137872
## 4 AKT_UP_MTOR_DN.V1_DN         134
## 5 AKT_UP_MTOR_DN.V1_DN       55326
## 6 AKT_UP_MTOR_DN.V1_DN         271</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">em &lt;-<span class="st"> </span><span class="kw">enricher</span>(gene, <span class="dt">TERM2GENE=</span>m_t2g)
em2 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)
<span class="kw">head</span>(em)</code></pre></div>
<pre><code>##                                            ID
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN
## CSR_LATE_UP.V1_UP           CSR_LATE_UP.V1_UP
## E2F1_UP.V1_UP                   E2F1_UP.V1_UP
##                                   Description GeneRatio
## RPS14_DN.V1_DN                 RPS14_DN.V1_DN    22/186
## GCNP_SHH_UP_LATE.V1_UP GCNP_SHH_UP_LATE.V1_UP    16/186
## VEGF_A_UP.V1_DN               VEGF_A_UP.V1_DN    15/186
## PRC2_EZH2_UP.V1_DN         PRC2_EZH2_UP.V1_DN    14/186
## CSR_LATE_UP.V1_UP           CSR_LATE_UP.V1_UP    12/186
## E2F1_UP.V1_UP                   E2F1_UP.V1_UP    12/186
##                          BgRatio       pvalue     p.adjust
## RPS14_DN.V1_DN         187/11250 4.072878e-13 6.923892e-11
## GCNP_SHH_UP_LATE.V1_UP 183/11250 5.657195e-08 4.808615e-06
## VEGF_A_UP.V1_DN        193/11250 6.903522e-07 3.911996e-05
## PRC2_EZH2_UP.V1_DN     195/11250 4.197805e-06 1.784067e-04
## CSR_LATE_UP.V1_UP      172/11250 2.766080e-05 9.404672e-04
## E2F1_UP.V1_UP          189/11250 6.963494e-05 1.972990e-03
##                              qvalue
## RPS14_DN.V1_DN         5.616284e-11
## GCNP_SHH_UP_LATE.V1_UP 3.900487e-06
## VEGF_A_UP.V1_DN        3.173198e-05
## PRC2_EZH2_UP.V1_DN     1.447138e-04
## CSR_LATE_UP.V1_UP      7.628557e-04
## E2F1_UP.V1_UP          1.600382e-03
##                                                                                                                                        geneID
## RPS14_DN.V1_DN         10874/55388/991/9493/1062/4605/9133/23397/79733/9787/55872/83461/54821/51659/9319/9055/10112/4174/5105/2532/7021/79901
## GCNP_SHH_UP_LATE.V1_UP                                      55388/7153/79733/6241/9787/51203/983/9212/1111/9319/9055/3833/6790/4174/3169/1580
## VEGF_A_UP.V1_DN                                                   8318/9493/1062/9133/10403/6241/9787/4085/332/3832/7272/891/23362/2167/10234
## PRC2_EZH2_UP.V1_DN                                               8318/55388/4605/23397/9787/55355/10460/81620/2146/7272/9212/11182/3887/24137
## CSR_LATE_UP.V1_UP                                                               55143/2305/4605/6241/11065/55872/983/332/2146/51659/9319/1580
## E2F1_UP.V1_UP                                                                  55388/7153/23397/79733/9787/2146/2842/9212/8208/1111/9055/3833
##                        Count
## RPS14_DN.V1_DN            22
## GCNP_SHH_UP_LATE.V1_UP    16
## VEGF_A_UP.V1_DN           15
## PRC2_EZH2_UP.V1_DN        14
## CSR_LATE_UP.V1_UP         12
## E2F1_UP.V1_UP             12</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(em2)</code></pre></div>
<pre><code>##                              ID     Description setSize
## IL2_UP.V1_DN       IL2_UP.V1_DN    IL2_UP.V1_DN     183
## RAF_UP.V1_DN       RAF_UP.V1_DN    RAF_UP.V1_DN     189
## ATF2_S_UP.V1_DN ATF2_S_UP.V1_DN ATF2_S_UP.V1_DN     182
## ATF2_UP.V1_DN     ATF2_UP.V1_DN   ATF2_UP.V1_DN     178
## IL15_UP.V1_DN     IL15_UP.V1_DN   IL15_UP.V1_DN     182
## LEF1_UP.V1_DN     LEF1_UP.V1_DN   LEF1_UP.V1_DN     188
##                 enrichmentScore       NES      pvalue
## IL2_UP.V1_DN         -0.4237496 -1.716666 0.001412429
## RAF_UP.V1_DN         -0.5521243 -2.247550 0.001414427
## ATF2_S_UP.V1_DN      -0.4456885 -1.806367 0.001416431
## ATF2_UP.V1_DN        -0.5029082 -2.033430 0.001416431
## IL15_UP.V1_DN        -0.4687839 -1.899973 0.001416431
## LEF1_UP.V1_DN        -0.4938672 -2.005243 0.001420455
##                  p.adjust     qvalues rank
## IL2_UP.V1_DN    0.0179863 0.009516559 2774
## RAF_UP.V1_DN    0.0179863 0.009516559 2035
## ATF2_S_UP.V1_DN 0.0179863 0.009516559 2531
## ATF2_UP.V1_DN   0.0179863 0.009516559 2055
## IL15_UP.V1_DN   0.0179863 0.009516559 2436
## LEF1_UP.V1_DN   0.0179863 0.009516559 1705
##                                   leading_edge
## IL2_UP.V1_DN    tags=39%, list=22%, signal=31%
## RAF_UP.V1_DN    tags=43%, list=16%, signal=37%
## ATF2_S_UP.V1_DN tags=39%, list=20%, signal=32%
## ATF2_UP.V1_DN   tags=38%, list=16%, signal=32%
## IL15_UP.V1_DN   tags=40%, list=19%, signal=33%
## LEF1_UP.V1_DN   tags=36%, list=14%, signal=32%
##                                                                                                                                                                                                                                                                                                                                                                                                                                            core_enrichment
## IL2_UP.V1_DN                                                  10858/5333/2977/79874/10014/7773/64798/26959/6588/579/777/5158/23361/9915/5087/64759/2535/6123/7010/51285/126353/80201/79841/1028/2322/4222/947/9187/225/9976/5592/54996/3202/55779/23148/5334/8605/57326/26115/54453/55766/64927/6591/55204/9940/57608/653483/9886/10279/55805/10628/901/9891/7102/27244/79921/4674/8292/5125/79843/6857/79940/23541/3357/50853/9863/79932/90362/55663/4250
## RAF_UP.V1_DN    221037/1960/51340/55105/10265/22996/9687/5357/1952/12/4602/9231/596/51454/57419/595/4256/23022/8777/7837/7042/3397/57613/323/1831/6451/54843/7358/2353/8773/6938/8991/64699/57007/23389/10560/7227/3485/79068/10769/4254/26018/3480/2674/23327/857/116039/6542/2690/1955/1363/26353/7033/5376/89927/51363/8821/2239/6947/4886/214/3487/3667/5157/54847/54898/7031/6505/57535/10451/18/771/80129/5174/5507/56521/8839/8614/5241/10551/57758
## ATF2_S_UP.V1_DN                                                              53616/55667/51279/4642/7130/7139/2/710/80352/4608/11145/881/2252/3399/22846/10687/12/1397/1301/3554/2619/26011/78987/9284/1306/6586/9627/4054/54813/115207/55890/26249/4803/11075/3400/79789/8527/2353/10580/79987/80760/5493/90865/3485/3751/2202/2199/10610/1047/1294/23492/3908/5364/658/5350/2121/2331/4330/9737/4982/57088/8490/9358/6424/1307/3708/125/1311/56521/10351
## ATF2_UP.V1_DN                                                                      9738/2252/8660/10687/5357/80201/12/22998/3554/2619/23556/8840/9284/399491/57419/5737/6586/9627/8863/10085/115207/55890/23015/629/3212/3400/50650/8527/2353/79987/80760/9668/1012/10614/5493/8553/23126/90865/3485/3751/2202/57212/2199/10610/3075/6387/22834/1909/10129/57161/5350/2121/2331/4330/2952/4982/57088/9358/6424/6038/51313/1311/56521/10351/5304/4239/79901
## IL15_UP.V1_DN                              64798/26959/6588/3784/777/113791/79981/64759/2535/957/6123/8038/51285/126353/55357/1470/2669/1028/4222/7450/225/1942/9976/29068/5592/54996/10272/6019/55779/80823/80215/5334/9639/57326/6403/26115/54453/57126/151011/57644/1831/51339/6591/55204/1393/1272/57608/79999/653483/9886/10279/55805/1047/10628/901/64123/27244/116039/79921/7041/54361/79843/6857/2239/6947/23541/55857/1811/3357/5104/56967/284266
## LEF1_UP.V1_DN                                                            22998/10325/23221/79170/9187/4300/51479/25956/7552/8644/11162/7168/1490/54463/65084/25837/80303/1809/3397/4208/11223/79762/6451/481/54795/51626/8991/5950/5627/64699/6414/10103/221078/9961/5874/56898/90865/57235/83989/5002/6653/56034/55314/85458/956/54502/4832/5783/1363/53832/54985/80221/84786/214/50853/51149/57088/5157/2947/79932/7031/6505/9338/25924/80736/3169/10551</code></pre>
<p>We can test with other collections, for example, using C3 to test whether the genes are up/down-regulated by sharing specific motif.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m_t2g &lt;-<span class="st"> </span><span class="kw">msigdbr</span>(<span class="dt">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="dt">category =</span> <span class="st">&quot;C3&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(gs_name, entrez_gene)
<span class="kw">head</span>(m_t2g)</code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   gs_name        entrez_gene
##   &lt;chr&gt;                &lt;int&gt;
## 1 AAACCAC_MIR140       10257
## 2 AAACCAC_MIR140          81
## 3 AAACCAC_MIR140          90
## 4 AAACCAC_MIR140        8754
## 5 AAACCAC_MIR140       11096
## 6 AAACCAC_MIR140         177</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">em3 &lt;-<span class="st"> </span><span class="kw">GSEA</span>(geneList, <span class="dt">TERM2GENE =</span> m_t2g)
<span class="kw">head</span>(em3)</code></pre></div>
<pre><code>##                                                            ID
## GGATTA_PITX2_Q2                               GGATTA_PITX2_Q2
## WTGAAAT_UNKNOWN                               WTGAAAT_UNKNOWN
## TGACATY_UNKNOWN                               TGACATY_UNKNOWN
## TTTGCAC_MIR19A_MIR19B                   TTTGCAC_MIR19A_MIR19B
## CAGTATT_MIR200B_MIR200C_MIR429 CAGTATT_MIR200B_MIR200C_MIR429
## WGTTNNNNNAAA_UNKNOWN                     WGTTNNNNNAAA_UNKNOWN
##                                                   Description
## GGATTA_PITX2_Q2                               GGATTA_PITX2_Q2
## WTGAAAT_UNKNOWN                               WTGAAAT_UNKNOWN
## TGACATY_UNKNOWN                               TGACATY_UNKNOWN
## TTTGCAC_MIR19A_MIR19B                   TTTGCAC_MIR19A_MIR19B
## CAGTATT_MIR200B_MIR200C_MIR429 CAGTATT_MIR200B_MIR200C_MIR429
## WGTTNNNNNAAA_UNKNOWN                     WGTTNNNNNAAA_UNKNOWN
##                                setSize enrichmentScore
## GGATTA_PITX2_Q2                    449      -0.3345811
## WTGAAAT_UNKNOWN                    458      -0.3214586
## TGACATY_UNKNOWN                    497      -0.3193965
## TTTGCAC_MIR19A_MIR19B              397      -0.3659786
## CAGTATT_MIR200B_MIR200C_MIR429     383      -0.3564722
## WGTTNNNNNAAA_UNKNOWN               420      -0.3201831
##                                      NES      pvalue
## GGATTA_PITX2_Q2                -1.486337 0.001209190
## WTGAAAT_UNKNOWN                -1.430554 0.001210654
## TGACATY_UNKNOWN                -1.424413 0.001213592
## TTTGCAC_MIR19A_MIR19B          -1.608477 0.001242236
## CAGTATT_MIR200B_MIR200C_MIR429 -1.561285 0.001245330
## WGTTNNNNNAAA_UNKNOWN           -1.413069 0.001253133
##                                  p.adjust    qvalues rank
## GGATTA_PITX2_Q2                0.02754988 0.01898976 2464
## WTGAAAT_UNKNOWN                0.02754988 0.01898976 1523
## TGACATY_UNKNOWN                0.02754988 0.01898976 2961
## TTTGCAC_MIR19A_MIR19B          0.02754988 0.01898976 4078
## CAGTATT_MIR200B_MIR200C_MIR429 0.02754988 0.01898976 3186
## WGTTNNNNNAAA_UNKNOWN           0.02754988 0.01898976 2754
##                                                  leading_edge
## GGATTA_PITX2_Q2                tags=26%, list=20%, signal=22%
## WTGAAAT_UNKNOWN                tags=18%, list=12%, signal=17%
## TGACATY_UNKNOWN                tags=30%, list=24%, signal=24%
## TTTGCAC_MIR19A_MIR19B          tags=44%, list=33%, signal=30%
## CAGTATT_MIR200B_MIR200C_MIR429 tags=34%, list=25%, signal=26%
## WGTTNNNNNAAA_UNKNOWN           tags=30%, list=22%, signal=24%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 core_enrichment
## GGATTA_PITX2_Q2                                                                                                                                                                                                                                                                                                                                         311/3199/4604/57556/8019/3321/57835/8929/6671/1760/10401/4430/9497/5530/6532/54894/23528/5228/5087/2295/6453/56171/3232/4921/3005/221037/56912/10265/10521/10497/23767/8322/37/23429/6561/4303/57082/845/94/51447/9922/6258/23243/2571/10253/4602/29119/55068/126393/7402/56675/5624/7716/7779/596/4300/57804/79923/1280/100506658/9627/3675/26037/2261/2804/65084/11030/3216/57326/140890/23255/5396/9854/1907/54843/57037/862/23452/55152/1848/8835/89795/57007/64221/5136/7227/10324/55273/6310/2202/56034/956/54502/51474/4254/6925/6097/744/443/23446/4306/4915/4487/25803/4081/4223/10472/388677/55107/347902/3708/10742/51313/4857/730/4036/3169
## WTGAAAT_UNKNOWN                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      2823/80762/50486/4628/9736/324/80070/51585/9627/8082/11211/4885/64112/7482/5334/79645/4734/80059/4208/11075/55084/2012/6709/6003/55629/8829/1581/54843/2353/862/727/10516/10580/1848/55204/5166/89795/2845/79776/9886/761/116496/4488/11096/10468/23677/9201/3075/6925/4121/443/55184/57332/5205/10129/2819/57161/51306/55812/3249/54361/54970/5744/26960/10186/10472/388677/4330/53829/26137/9079/8404/36/3572/3479/1602/23090/81563/2167/8614/57758/4969
## TGACATY_UNKNOWN                                                                                                                                                                             800/5168/55691/5789/11080/6400/55512/1112/9604/29760/4638/3218/27314/2065/4286/9459/4026/7026/4335/2263/53353/54329/7082/10446/5930/8567/51334/10174/8626/687/3321/83478/6588/5441/8929/7881/9846/23328/8924/220988/146057/55909/4661/29116/23405/94134/56171/63898/4131/4776/80000/627/9748/2908/11138/23767/10290/5295/1288/9445/10231/29/3373/2669/5191/4211/5813/25945/7716/596/4300/2078/3931/8633/9627/2872/3213/80021/2261/11194/3223/4212/8848/5837/6764/23037/57326/7106/590/7042/4208/55084/50650/80052/6450/323/481/8436/2353/10370/862/9120/727/79899/23452/1293/10580/60481/54838/55204/5166/2845/81603/6876/8522/1592/5577/3489/6310/2202/29951/27239/55821/9499/1501/57161/6571/4035/8821/5744/90627/4922/55800/4223/10472/23303/1264/54587/9590/9576/185/4675/576/1287/9358/3479/5348/776/771/79689
## TTTGCAC_MIR19A_MIR19B          1605/659/10318/4299/6249/81609/57194/9686/10425/9839/23013/22936/989/51742/4154/55625/51460/51592/11127/3976/54980/9751/56937/6045/4204/27252/4862/27236/2115/5869/152006/57698/57018/23032/56145/10150/55054/4090/25852/10618/2060/7107/10771/7071/9044/10904/9752/6431/8473/83452/392/10395/6777/51230/22889/23236/55652/158471/6595/905/56146/4026/79665/5932/115/8325/546/10154/9655/23499/54521/26959/23612/3321/56929/54778/26060/56848/9098/107/10420/23001/29922/9759/29116/23405/94134/2295/6453/83660/51719/8654/11138/2186/23341/8038/860/2004/9256/6548/23621/23503/29/394/220594/831/92689/55082/54014/1901/23111/7552/27303/1490/51454/57419/66008/2318/6252/7248/4929/23041/9900/55727/91694/22863/1389/54796/3400/9649/23253/54453/7048/22862/22905/23365/79899/5412/60481/54838/89795/5793/23389/7227/6310/3625/26018/23492/2099/23116/744/775/90355/388/23414/11069/6857/26960/55638/10472/2152/2697/3131/7786/55107/9863/23261/3479/3708/10742/2018/4036/2066
## CAGTATT_MIR200B_MIR200C_MIR429                                                                                                                                                                                                                                                  31/23291/5179/57092/9969/9865/8473/9644/10395/6383/5569/1392/4008/9647/11041/8613/6405/4884/8491/9706/2263/6567/53353/64766/5310/546/5128/55145/54469/200576/8434/7003/687/3915/1456/3321/10499/23189/26060/55578/1998/51421/22864/10484/10140/23001/4661/253782/9807/83660/95681/1960/2908/51719/2186/22846/108/9710/80205/2104/9522/7320/23243/23258/4602/129642/2669/51439/7716/54014/23047/5099/2823/3340/1942/7552/1843/100506658/8204/10979/79618/23015/8848/4692/5334/57515/26249/6563/23122/6196/4077/2494/10370/3953/51339/79365/54838/89795/8076/5577/3751/6310/1009/29994/10769/1501/57509/3899/2273/23116/4908/27244/775/5783/65055/4035/23414/2737/5744/6857/55800/10186/3131/9828/6935/4982/219654/1602/3708/4857
## WGTTNNNNNAAA_UNKNOWN                                                                                                                                                                                                                                                                                              9459/64132/4026/7026/79665/29072/83604/5218/7020/64766/54329/2114/10124/79742/1105/51279/4601/1298/51747/5071/8929/10681/6671/79729/8843/8609/23528/10211/94134/10206/4131/79960/80000/221037/627/2252/10521/5295/7010/9710/51780/3221/23429/2104/26054/25876/2571/22902/23258/63971/29119/604/23221/55068/460/4222/5099/2078/4216/9315/27303/64388/3202/54819/58190/3223/4256/4929/4212/3216/140890/79641/23635/167/54796/6003/54453/9854/553/80267/22862/2353/23365/862/2638/1848/54838/8835/1592/3489/3778/51673/1657/477/9073/29951/4313/4337/10468/9369/79667/9499/6925/3480/4908/744/1462/51309/5101/4741/51363/25803/213/8800/10472/26137/9079/1634/4128/23090/4857/652/3169/7021/4969</code></pre>
<!--


# DAVID functional analysis

[clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) provides enrichment and GSEA analysis with GO, KEGG, DO and Reactome pathway supported internally, some user may prefer GO and KEGG analysis with DAVID[@huang_david_2007] and still attracted by the visualization methods provided by [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler)[@paranjpe_genome_wid_2013]. To bridge the gap between DAVID and clusterProfiler, we implemented `enrichDAVID`. This function query enrichment analysis result from DAVID webserver via [RDAVIDWebService](https://www.bioconductor.org/packages/RDAVIDWebService)[@fresno_rdavidwebservice_2013] and stored the result as an `enrichResult` instance, so that we can use all the visualization functions in [clusterProfiler](https://www.bioconductor.org/packages/clusterProfiler) to visualize DAVID results. `enrichDAVID` is fully compatible with `compareCluster` function and comparing enrichment results from different gene clusters is now available with DAVID.


```r
david <- enrichDAVID(gene = gene,
                     idType = "ENTREZ_GENE_ID",
                     listType = "Gene",
                     annotation = "KEGG_PATHWAY",
                     david.user = "clusterProfiler@hku.hk")
```

DAVID Web Service has the following limitations:

+ A job with more than 3000 genes to generate gene or term cluster report will not be handled by DAVID due to resource limit.
+ No more than 200 jobs in a day from one user or computer.
+ DAVID Team reserves right to suspend any improper uses of the web service without notice.

For more details, please refer to [http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html](http://david.abcc.ncifcrf.gov/content.jsp?file=WS.html).

As user has limited usage, please [register](http://david.abcc.ncifcrf.gov/webservice/register.htm) and use your own user account to run `enrichDAVID`.


-->

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
